test/component_test.go

```
@@ -0,0 +1,113 @@
+					Resource  interface{} `json:"Resource"` // Changed to interface{} to accommodate array
+					Condition struct {
+						StringEquals    map[string]string `json:"StringEquals,omitempty"`
+						StringNotEquals map[string]string `json:"StringNotEquals,omitempty"`
+						Null            map[string]string `json:"Null,omitempty"`
+						Bool            map[string]bool   `json:"Bool,omitempty"` // Added Bool for new condition
+					} `json:"Condition"`
+				} `json:"Statement"`
+			}
+
+			var policy BucketPolicy
+			json.Unmarshal([]byte(policyString), &policy)
```

golangci-lint (1.62.2)

Error return value of `json.Unmarshal` is not checked

(errcheck)